src/application.ts

```diff
@@ -1,5 +1,6 @@
 import * as express from 'express'
 import {EventEmitter} from 'promise-events'
+import {ApplicationFunction} from '.'
```

Comment:
@bkeepers I went with `ApplicationFunction`. Feel free to push to this branch if you (or anyone else) has a better name; otherwise I'll go ahead and merge this within the next day or so.

Reply:
I don't love it, but I don't hate it any more than any other option. 👍 for going with that for now.

Reply:
I'll leave it here for a little in case someone smarter than us has a brilliant idea.